### PR TITLE
✨ #253: Data Mapper route to get invite by ID

### PR DESCRIPTION
**Pull request by alyssadsouza**
## Description of Changes

Implements the `GET /invites/:inviteId` route and the services it calls in Data Mapper, which fetches a created Clinician Invite. Includes error handling, validation, and logging. This completes the Clinician Invite flow as of now (this route will be used in the registration flow by Consent API).

### Data Mapper
- Added the endpoint to retrieve the invite, with error handling for `404: NotFoundError` (invite does not exist) and `500: ServerError` (unexpected error)
- Added the `getInviteConsentData` service which makes a request to Consent DAS to fetch the Consent invite
- Added the `getInvitePiData` service which makes a request to PI DAS to fetch the PI invite
- Added the `getInvite` service which calls `getInvitePiData` and then `getInviteConsentData` to fetch the Consent and PI invites, and then combines and returns both objects

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [x] PR feedback has been addressed **if applicable**
- [x] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/data-mapper/src/services/das/consent.ts [hunk 1]
```diff
@@ -26,9 +26,46 @@ import { getAppConfig } from '../../config.js';
...
+ * @returns ClinicianInvite object from Consent DB
+ */
+export const getInviteConsentData = async (
+	inviteId: string,
+): Promise<Result<ConsentClinicianInviteResponse, GetInviteFailureStatus>> => {
+	const { consentDasUrl } = getAppConfig();
+	try {
+		const { data } = await axiosClient.get(urlJoin(consentDasUrl, 'clinician-invites', inviteId));
+
+		const invite = ConsentClinicianInviteResponse.safeParse(data); // converts all nulls to undefined
```
Comment (mistryrn) [inline apps/data-mapper/src/services/das/consent.ts:45]: 👍 thanks for the comment here!

**Review by mistryrn — APPROVED** [review]
Looks good! 🚀

File: apps/data-mapper/src/services/das/consent.ts [hunk 2]
```diff
@@ -26,9 +26,46 @@ import { getAppConfig } from '../../config.js';
...
 import axiosClient from '../axiosClient.js';
 import { CreateInviteFailureStatus } from '../create.js';
+import { GetInviteFailureStatus } from '../search.js';
 
 const logger = serviceLogger.forModule('ConsentClient');
 
+/**
+ * Makes request to Consent DAS to fetch a Clinician Invite
+ * @param inviteId
+ * @returns ClinicianInvite object from Consent DB
```
Comment (anncatton) [inline apps/data-mapper/src/services/das/consent.ts:36]: can specify the `return` to be `ConsentClinicianInviteResponse`

File: apps/data-mapper/src/services/das/pi.ts [hunk 3]
```diff
@@ -26,9 +26,46 @@ import { getAppConfig } from '../../config.js';
...
 import axiosClient from '../axiosClient.js';
 import { CreateInviteFailureStatus } from '../create.js';
+import { GetInviteFailureStatus } from '../search.js';
 
 const logger = serviceLogger.forModule('PIClient');
 
+/**
+ * Makes request to PI DAS to fetch a Clinician Invite
+ * @param inviteId
+ * @returns ClinicianInvite object from PI DB
```
Comment (anncatton) [inline apps/data-mapper/src/services/das/pi.ts:36]: can specify `return` type to be `PIClinicianInviteResponse`

File: apps/data-mapper/src/services/search.ts [hunk 4]
```diff
@@ -54,3 +61,45 @@ export const getParticipant = async (participantId: string): Promise<any> => {
...
 };
+
+export type GetInviteFailureStatus = 'SYSTEM_ERROR' | 'INVITE_DOES_NOT_EXIST';
+/**
+ * Fetches clinician invite in PI DAS first by inviteId,
+ * then uses the same inviteId to get the corresponding invite in Consent DAS
+ * Combines both invite objects and returns as Clinician Invite
+ * @async
+ * @param inviteId
+ * @returns Clinician Invite
```
Comment (anncatton) [inline apps/data-mapper/src/services/search.ts:72]: specify `ClinicianInviteResponse` as return type

**Review by anncatton — COMMENTED** [review]
Great! Just a couple smol nitpicks 😛

**Review by anncatton — APPROVED** [review]
thanks @alyssadsouza !

**Pull request merged**